### DISCUSSION TITLE
I am not able to use order in my query please give me solution

### DISCUSSION THREAD

**Discussion by nishantbhavasar** [supabase/supabase#19249] [Questions] [answered]
i have created Two tables
1) user
id primary key
name
phone
email
address

2) auth_users
id primarykey
user_id reference of user table id (foreign key to user.id)
phone_verified_at timestamp
email_verified_at timestamp

now below is my query 

supabase
    .from("users")
    .select(
      "id,name,email,phone,roles!inner(role_sc),auth_users(id,phone_confirmed_at,email_confirmed_at)",
      {
        count: "exact",
      }
    )
    .order("phone_verified_at",{ascending:false,foreignTable:"auth_users"})
    .range(from,to)
    
 but this order by phone_verified_at not working properly i am getting response properly as aspected but the order not appling on my result 
 please give me solution 
 
 thank you

**Comment by GaryAustin1** [comment]
You may need phone_verified_at in the select for that table.
Also the order by you are using will order the embedded (foreign) table NOT the parent table by the results of that table.
If you want to order the embedded table you use .order(auth_users(phone_verified_at)).
See "order by related tables" here https://supabase.com/blog/postgrest-11-prerelease

**Reply by nishantbhavasar** [reply]
@GaryAustin1  please help to resolve this issue

**Reply by GaryAustin1** [reply, marked as answer]
Per your error below there is no way to order in the API on a foreign relation that has multiple rows.  I'm not sure how you would expect an order to occur if there are multiple entries.   
If auth_users has only one entry per user then you can make the user_id column UNIQUE.  This would then make it a one to one relationship and solve the issue I believe.

Otherwise at some point it is just easier to set up a Postgres view.

**Reply by nishantbhavasar** [reply]
@GaryAustin1 thankyou so much for helping me in this issue ,
This work after creating user_id UNIQUE in auth_users table 
thank you again

**Comment by nishantbhavasar** [comment]
thank's for your answer but still i am facing issue like this only work with one-to-one or many-to-one but what does do in situation where i have one-to-many relationship b/w tables like my auth_users table has user_id which is a reference of user table primary key but in y user table i don't have any fk for connecting auth_user table so it's like one to many relationship or not working auth_user(phone_verified_at) in my query or i am getting below error

{
    "code": "PGRST118",
    "details": "'users' and 'auth_users' do not form a many-to-one or one-to-one relationship",
    "hint": null,
    "message": "A related order on 'auth_users' is not possible"
}

please give me solution
